aptos-move/aptos-release-builder/src/simulate.rs

```diff
@@ -0,0 +1,601 @@
+        let deserializer_config = aptos_prod_deserializer_config(&features);
+
+        patch_create_signer(&state_view, &deserializer_config)?;
+        // If the script is the last step of the proposal, it MUST NOT have a next execution hash.
+        // Set the boolean flag to true to use a modified patch to catch this.
+        let forbid_next_execution_hash = script_idx == scripts.len() - 1;
+        patch_resolve_multistep_proposal(
+            &state_view,
+            &deserializer_config,
+            forbid_next_execution_hash,
+        )?;
+
+        let script_name = script_path.file_name().unwrap().to_string_lossy();
+        println!("    {}", script_name);
+
+        // Create a new VM to ensure the loader is clean.
+        // The warm vm cache also needs to be explicitly flushed as it cannot detect the
+        // patches we performed.
+        flush_warm_vm_cache();
+        let vm = AptosVM::new(&state_view);
+        let log_context = AdapterLogSchema::new(state_view.id(), 0);
+        let resolver = state_view.as_move_resolver();
+        let (_vm_status, vm_output) = vm.execute_user_transaction(
+            &resolver,
+            &account
+                .account()
+                .transaction()
+                .script(Script::new(script_blob, vec![], vec![
+                    TransactionArgument::U64(DUMMY_PROPOSAL_ID), // dummy proposal id, ignored by the patched function
+                ]))
+                .chain_id(chain_id.chain_id())
+                .sequence_number(script_idx as u64)
+                .gas_unit_price(gas_params.vm.txn.min_price_per_gas_unit.into())
+                .max_gas_amount(100000)
+                .ttl(u64::MAX)
+                .sign(),
+            &log_context,
+        );
+        // TODO: ensure all scripts trigger reconfiguration.
+
+        let txn_output = vm_output.try_materialize_into_transaction_output(&resolver)?;
+        let txn_status = txn_output.status();
+        match txn_status {
+            TransactionStatus::Keep(ExecutionStatus::Success) => {
+                println!("        Success")
+            },
+            TransactionStatus::Keep(ExecutionStatus::MoveAbort { code, .. })
+                if *code == MAGIC_FAILED_NEXT_EXECUTION_HASH_CHECK =>
+            {
+                bail!("the last script has a non-zero next execution hash")
+            },
+            _ => {
+                println!(
+                    "{}",
+                    format!("{:#?}", txn_status)
+                        .lines()
+                        .map(|line| format!("        {}", line))
+                        .collect::<Vec<_>>()
+                        .join("\n")
+                );
+                bail!("failed to execute governance script: {}", script_name)
+            },
+        }
+
+        let (write_set, _events) = txn_output.into();
+        state_view.apply_write_set(write_set);
+    }
+
+    println!("All scripts succeeded!");
```

Comment:
We need to check there's no pending script hash there. Can you add the check or at least add a todo here? I think it would be important for our release safety.

Reply:
I think I've implemented this check, no?

Reply:
Oh nvm. I thought you didn't.